### PR TITLE
#3161 Add tests to verify devTool dialog works well

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3161)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/cypress/e2e/devToolDialog.test.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  .substring(0, 2)
+  .toUpperCase();
+
+describe('Topics tab', () => {
+  before(async () => await deleteAllServices());
+
+  it('empty topic list', () => {
+    // close the quickstart dialog
+    cy.visit('/')
+      .get('.close-button')
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/devToolDialog.test.js]: could we use `findByText` ~~or `getByText` here? (sorry, i couldn't recall the exact method names as I haven't touched on cypress like for years...)~~ that is a more robust selector and reflect directly with our UI changes not "implementation details"

File: ohara-manager/client/cypress/e2e/devToolDialog.test.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    cy.findByTitle('Developer Tools')
+      .click()
+      .findByText(/topics/i)
+      .should('exist');
+
+    // check the topic Select component
+    cy.get('.MuiSelect-select')
+      .should('have.length', 1)
+      .click()
+      .get('ul')
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/devToolDialog.test.js]: For HTML element that doesn't have text content, we could use the `data-testid` instead

File: ohara-manager/client/cypress/e2e/devToolDialog.test.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+          .click();
+      })
+      .end();
+  });
+});
+
+describe('Logs tab', () => {
+  before(async () => await deleteAllServices());
+
+  it.only('normal test', () => {
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/devToolDialog.test.js]: Please remove the only keyword. And this should make the CI build fail as it's part of the linter rule

File: ohara-manager/client/cypress/e2e/devToolDialog.test.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        cy.window()
+          .its('open')
+          .should('be.called');
+
+        // close button
+        cy.get('.items span>button:visible')
+          .eq(3)
+          .click();
+      })
+      .end();
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/devToolDialog.test.js]: Not familiar with `.end()`... is it necessary to put it here?

File: ohara-manager/client/cypress/support/e2eCommands.js [hunk 5]
```diff
@@ -15,3 +15,188 @@
...
+        const topic = {
+          name: generate.serviceName({ prefix: 'topic' }),
+          group: hashByGroupAndName(workspaceGroup, workspaceName),
+          nodeNames: [node.hostname],
+          brokerClusterKey: {
+            name: broker.name,
+            group: broker.group,
+          },
+          tags: {
+            type: 'shared',
```
Comment (konekoya) [inline ohara-manager/client/cypress/support/e2eCommands.js]: I've discussed this with @eechih this morning, and it will be changed to:

```js
tags: {
    isShared: boolean
}
```

File: ohara-manager/client/cypress/support/e2eCommands.js [hunk 6]
```diff
@@ -15,3 +15,188 @@
...
+        group: workerGroup,
+      },
+    };
+
+    const connectorRes = await connectorApi.create(connector);
+    if (connectorRes.errors) throw new Error(connectorRes.title);
+
+    await connectorApi.start(connector);
+
+    // sleep 10 seconds for perf connector writes some data to topic
```
Comment (konekoya) [inline ohara-manager/client/cypress/support/e2eCommands.js]: would rephrase this to:

```suggestion
    // sleep 10 seconds for perf connector to write some data to topic
```

:)

File: ohara-manager/client/cypress/utils.js [hunk 7]
```diff
@@ -100,7 +100,7 @@ export const deleteAllServices = async () => {
   const connects = connectRes.data;
   // we don't care the execute order of each individual connect was done or not.
   // Using Promise.all() to make sure all connects were stopped & deleted.
-  await Promise.all(connects.map(connect => connectorApi.stop(connect)));
+  await Promise.all(connects.map(connect => connectorApi.forceStop(connect)));
```
Comment (konekoya) [inline ohara-manager/client/cypress/utils.js]: didn't know this is still a thing 👍

File: ohara-manager/client/src/api/brokerApi.js [hunk 8]
```diff
@@ -80,6 +80,20 @@ export const update = async params => {
   return result;
 };
 
+export const forceStop = async params => {
```
Comment (konekoya) [inline ohara-manager/client/src/api/brokerApi.js]: So the API is specifically for end to end tests only?

File: ohara-manager/client/cypress/e2e/devToolDialog.test.js (hunk 3 above)
Comment (saivirtue) [reply ohara-manager/client/cypress/e2e/devToolDialog.test.js]: My mistake :(

File: ohara-manager/client/cypress/e2e/devToolDialog.test.js (hunk 4 above)
Comment (saivirtue) [reply ohara-manager/client/cypress/e2e/devToolDialog.test.js]: code style. To avoid further testing using the previous DOM element :)

File: ohara-manager/client/cypress/support/e2eCommands.js (hunk 5 above)
Comment (saivirtue) [reply ohara-manager/client/cypress/support/e2eCommands.js]: 👍

File: ohara-manager/client/src/api/brokerApi.js (hunk 8 above)
Comment (saivirtue) [reply ohara-manager/client/src/api/brokerApi.js]: For now, I think so.
It's a "harmful" function and we should avoid using it in our UI :)

File: ohara-manager/client/cypress/e2e/devToolDialog.test.js (hunk 1 above)
Comment (saivirtue) [reply ohara-manager/client/cypress/e2e/devToolDialog.test.js]: Let me try it.

Comment (saivirtue) [reply ohara-manager/client/cypress/e2e/devToolDialog.test.js]: The `.get` function had been replaced by `findByText, findByTitle, findByTestId` as well :)

File: ohara-manager/client/cypress/e2e/devToolDialog.test.js (hunk 2 above)
Comment (saivirtue) [reply ohara-manager/client/cypress/e2e/devToolDialog.test.js]: ditto

**Comment by saivirtue** [comment]
retry manager-e2e

**Comment by saivirtue** [comment]
retry manager-e2e

**Comment by saivirtue** [comment]
retry manager-e2e

**Comment by saivirtue** [comment]
retry manager-e2e

**Comment by saivirtue** [comment]
retry manager-e2e

**Comment by saivirtue** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

File: ohara-manager/client/cypress/e2e/devToolDialog.test.js [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+    // check the topic Select component
+    cy.findByTitle('Select topic')
+      .should('have.length', 1)
+      .click()
+      .get('ul')
+      .should('be.empty');
+  });
+
+  it('normal test', () => {
```
Comment (konekoya) [inline ohara-manager/client/cypress/e2e/devToolDialog.test.js]: We should probably use a more descriptive name for the test

File: ohara-manager/client/cypress/support/retryOnFail.js [hunk 10]
```diff
@@ -19,5 +19,5 @@ const isE2eTest = Cypress.config('integrationFolder').includes('e2e');
 // Enable cypress to retry on fail end-to-end test
 if (isE2eTest) {
   require('cypress-plugin-retries');
-  Cypress.env('RETRIES', 3);
+  // Cypress.env('RETRIES', 3);
```
Comment (konekoya) [inline ohara-manager/client/cypress/support/retryOnFail.js]: Do remember to enable this on CI as this increase the rate of passing builds

Comment (saivirtue) [reply ohara-manager/client/cypress/support/retryOnFail.js]: Hmm...Thanks for reminding

**Comment by saivirtue** [comment]
retry manager-e2e

**Comment by saivirtue** [comment]
Since v0.9, we have a passing e2e tests now 💪
